tests/test_class.py

```diff
@@ -377,7 +377,8 @@ class PyDog(m.Dog):
         refcount_3 = getrefcount(cls)
 
         assert refcount_1 == refcount_3
-        assert refcount_2 > refcount_1
+        # Free-threaded Python uses UINT32_MAX for immortal objects.
+        assert (refcount_2 > refcount_1) or (refcount_2 == refcount_1 == 2**32 - 1)
```

Comment:
It would be much more self-documenting and future-proof to be organized about this from the start. E.g. by adding

```
#define PYBIND11_REFCNT_IMMORTAL UINT32_MAX
```

in detail/common.h and then expose that in tests/pybind11_tests.cpp (`PYBIND11_MODULE`) similar to `PYBIND11_INTERNALS_ID`.

Reply:
Okay, I did that, though I made it a tests-only thing, not something we publicly provide. I don't think it's really guaranteed, the general rule in Python is that "large" refcounts should be considered immortal. Where large is some value >= 2. https://docs.python.org/3/c-api/refcounting.html#c.Py_REFCNT